lib.sh

```diff
@@ -91,16 +93,63 @@ function get_package_name_ver {
 # Arguments:
 #   The comma and/or space delimited list of packages.
 # Returns:
-#   Sorted list of space delimited packages.
+#   Sorted list of space delimited package name=version pairs.
 ###############################################################################
 function get_normalized_package_list {
-  # Remove commas, and block scalar folded backslashes.
-  local stripped=$(echo "${1}" | sed 's/[,\]/ /g')
-  # Remove extraneous spaces at the middle, beginning, and end.
-  local trimmed="$(\
-    echo "${stripped}" \
-    | sed 's/\s\+/ /g; s/^\s\+//g; s/\s\+$//g')"
-  echo ${trimmed} | tr ' ' '\n' | sort | tr '\n' ' '
+  # Remove commas, and block scalar folded backslashes,
+  # extraneous spaces at the middle, beginning and end
+  # then sort.
+  packages=$(echo "${1}" | \
+    sed 's/[,\]/ /g; s/\s\+/ /g; s/^\s\+//g; s/\s\+$//g' | \
+    sort -t' ')
+
+  # Validate package names and get versions.
+  log_err "resolving package versions..."
```

Comment:
This should be log I believe. Also upper case R.

Reply:
This can't be a log as it would interfere with the returned value unfortunately.